src/uint/neg.rs

```diff
@@ -12,7 +17,7 @@ impl<const LIMBS: usize> Uint<LIMBS> {
             carry = r >> Limb::BITS;
             i += 1;
         }
-        Uint::new(ret)
+        (Uint::new(ret), carry as Word)
```

Comment:
btw. to fix that you would have to return `!0 + carry` as carry (which again is consistent with how you compute every limb value). I wanted to just explain above why is that a case.

Reply:
done!